packages/dd-trace/test/opentelemetry/metrics.spec.js

```diff
@@ -0,0 +1,1035 @@
+
+    if (httpStub) {
+      httpStub.restore()
+      httpStub = null
+    }
+    sinon.restore()
+  })
+
+  describe('Basic Functionality', () => {
+    it('exports counter metrics', (done) => {
+      const validator = mockOtlpExport((decoded) => {
+        const metrics = decoded.resourceMetrics[0].scopeMetrics[0].metrics
+        assert.strictEqual(metrics.length, 1)
+        assert.strictEqual(metrics[0].name, 'requests')
+        assert.strictEqual(metrics[0].sum.isMonotonic, true)
+        assert.strictEqual(metrics[0].sum.dataPoints[0].asDouble, 10.3)
+      })
+
+      setupTracer()
+      const meter = metrics.getMeter('app')
+      const counter = meter.createCounter('requests')
+      counter.add(5.1)
+      counter.add(5.2)
+
+      setTimeout(() => { validator(); done() }, 150)
+    })
+
+    it('exports histogram metrics', (done) => {
+      const validator = mockOtlpExport((decoded) => {
+        const histogram = decoded.resourceMetrics[0].scopeMetrics[0].metrics[0]
+        assert.strictEqual(histogram.name, 'duration')
+        assert.strictEqual(histogram.histogram.dataPoints[0].count, 1)
+        assert.strictEqual(histogram.histogram.dataPoints[0].sum, 100)
+      })
+
+      setupTracer()
+      const meter = metrics.getMeter('app')
+      meter.createHistogram('duration').record(100)
+
+      setTimeout(() => { validator(); done() }, 150)
+    })
+
+    it('ignores negative values and callback errors', (done) => {
+      let validated = false
+      const validator = mockOtlpExport((decoded) => {
+        const allMetrics = decoded.resourceMetrics[0].scopeMetrics[0].metrics
+        const histogram = allMetrics.find(m => m.name === 'size')
+        const counter = allMetrics.find(m => m.name === 'requests')
+        const gauge = allMetrics.find(m => m.name === 'memory')
+
+        // Only validate the export that has all three metrics (first export)
+        if (histogram && counter && gauge && !validated) {
+          assert.strictEqual(histogram.histogram.dataPoints[0].count, 2)
+          assert.strictEqual(histogram.histogram.dataPoints[0].sum, 300)
+          assert.strictEqual(counter.sum.dataPoints[0].asInt, 15)
+          assert.strictEqual(gauge.gauge.dataPoints[0].asInt, 100)
+          validated = true
+        }
+      })
+
+      setupTracer()
+      const meter = metrics.getMeter('app')
+
+      const hist = meter.createHistogram('size')
+      hist.record(100)
+      hist.record(-50)
+      hist.record(200)
+      hist.record(-100)
+
+      const counter = meter.createCounter('requests')
+      counter.add(10)
+      counter.add(-5)
+      counter.add(5)
+
+      const gauge = meter.createObservableGauge('memory')
+      gauge.addCallback(() => { throw new Error('Callback error') })
+      gauge.addCallback((result) => result.observe(100))
+
+      setTimeout(() => {
+        assert(validated, 'Should have validated an export with all metrics')
+        validator()
+        done()
+      }, 150)
```

Comment:
Instead of using a timeout, it would be possible to just call `done()` from inside of the `validator` method. That way it is simpler and faster.

The same applies for all other methods with that pattern.

It is also more reliable, due to not being dependent on the timing anymore.

Or even better: the `mockOtlpExport` method returns a promise. That way you can trigger the methods sync and just return that promise afterwards. That would be very clean.